src/client/common/alias_dict.cpp

```diff
@@ -97,37 +142,166 @@ bool mp::AliasDict::remove_alias(const std::string& alias)
     return false;
 }
 
-std::vector<std::string> mp::AliasDict::remove_aliases_for_instance(const std::string& instance)
+bool mp::AliasDict::remove_context(const std::string& context)
+{
+    if (aliases.erase(context) > 0)
+    {
+        modified = true;
+
+        if (active_context == context)
+            set_active_context(default_context_name);
+
+        return true;
+    }
+
+    return false;
+}
+
+// This function removes the context called as the instance, and then it iterates through all the remaining contexts
+// to see if there are aliases defined for the instance.
+std::vector<std::pair<std::string, std::string>> mp::AliasDict::remove_aliases_for_instance(const std::string& instance)
```

Comment:
`std::vector<std::pair<std::string, std::string>>` sure is a mouth full :slightly_smiling_face: 

Since this is used in a few places, perhaps this is worthy of a `typedef`?

Reply:
Good idea! I created a `typedef` named `ContextAliasPair`, in order to be clear of what is it. Feel free to suggest a better name.

Reply:
I think that name is clear.  Thanks!